<launch>
  <arg name="color_width"         default="640"/>
  <arg name="color_height"        default="480"/>
  <arg name="enable_color"        default="true"/>
  <arg name="color_fps"           default="30"/>
  <arg name="tf_prefix"           default=""/>
  <arg name="color_frame_id"      default="$(arg tf_prefix)_color_frame"/>
  <arg name="color_optical_frame_id"    default="$(arg tf_prefix)_color_optical_frame"/>

  <group ns="stereo/camera0/image" >
    <rosparam param="disable_pub_plugins">
      - "image_transport/compressed"
      - "image_transport/compressedDepth"
      - "image_transport/theora"
    </rosparam>
  </group>

  <group ns="stereo/camera1/image" >
    <rosparam param="disable_pub_plugins">
      - "image_transport/compressed"
      - "image_transport/compressedDepth"
      - "image_transport/theora"
    </rosparam>
  </group>
	
  <group ns="stereo">
    <param name="global_time" type="bool" value="true" />
    <param name="sync_size" type="int" value="30" />
    <param name="enable_emitter" type="bool" value="false" />

    <param name="frame_rate" type="int" value="30" />
    <param name="format" type="str" value="Y8" />
    <param name="width" type="int" value="640" />
    <param name="height" type="int" value="480" />
    <param name="exposure" type="double" value="10000" />
  </group>

  <group ns="camera">
    <param name="global_time"              type="bool" value="true" />
    <param name="sync_size" type="int" value="30" />
    <param name="color_width"              type="int"  value="$(arg color_width)"/>
    <param name="color_height"             type="int"  value="$(arg color_height)"/>
    <param name="color_format"             type="str"  value="RGB8" />
    <param name="enable_color"             type="bool" value="$(arg enable_color)"/>
    <param name="color_fps"                type="int"  value="$(arg color_fps)"/>
    <param name="color_frame_id"           type="str"  value="$(arg color_frame_id)"/>
    <param name="color_optical_frame_id"   type="str"  value="$(arg color_optical_frame_id)"/>
  </group>
  <node pkg="rs4se"
      type="intel_d435i"
      name="intel_d435i"
      required="true"
      output="screen">
  </node>
</launch>
